src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                    // var a = new [] { from a in b select a };                    
+                    case SyntaxKind.ArrayInitializerExpression:
+                    // var a = from a in b select a
+                    case SyntaxKind.EqualsValueClause:
+                    // a = from a in b select a
+                    case SyntaxKind.SimpleAssignmentExpression:
+                    // (from a in b select a)?.ToList();
+                    case SyntaxKind.ConditionalAccessExpression:
+                    // new new List<IEnumerable<int>> { from a in q select a * a };
+                    case SyntaxKind.CollectionInitializerExpression:
+                    // new new List<int>(from a in q select a * a );
+                    case SyntaxKind.Argument:
+                        return TryReplaceWithLocalFunction(source, parent, out documentUpdateInfo);
+                    default:
+                        documentUpdateInfo = default;
+                        return false;
+                }
+            }
+
+            private bool TryConvertIfInMemberAccessExpression(
+               QueryExpressionSyntax source,
+               MemberAccessExpressionSyntax memberAccessExpression,
+               out DocumentUpdateInfo documentUpdateInfo)
+            {
+                if (memberAccessExpression.Parent is InvocationExpressionSyntax invocationExpression)
+                {
+                    // This also covers generic names (i.e. with type arguments) like 'ToList<int>'. 
+                    // The ValueText is still just 'ToList'. 
+                    switch (memberAccessExpression.Name.Identifier.ValueText)
+                    {
+                        case nameof(Enumerable.ToList):
+                            return TryConvertIfInToListInvocation(source, invocationExpression, out documentUpdateInfo);
+                        case nameof(Enumerable.Count):
+                            return TryConvertIfInCountInvocation(source, invocationExpression, out documentUpdateInfo);
+                        default:
+                            return TryReplaceWithLocalFunction(source, memberAccessExpression, out documentUpdateInfo);
+                    }
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInCountInvocation(
+                QueryExpressionSyntax source,
+                InvocationExpressionSyntax invocationExpression,
+                out DocumentUpdateInfo documentUpdateInfo)
+            {
+                if (_semanticModel.GetSymbolInfo(invocationExpression, _cancellationToken).Symbol is IMethodSymbol methodSymbol &&
+                    methodSymbol.Parameters.Length == 0 && methodSymbol.ReturnType?.SpecialType == SpecialType.System_Int32)
+                {
+                    // before var count = (from a in b select a).Count();
```

Comment:
>var count = (from a in b select a).Count(); [](start = 30, length = 43)

Where do we check that this is the case we are dealing with? What if ```(from a in b select a).Count()``` is an argument to a call? Do we have a test like that?

Reply:
Could you please point to the test?

---
In reply to: [176929133](https://github.com/dotnet/roslyn/pull/25362#discussion_r176929133) [](ancestors = 176929133)

Reply:
just added it. will be available in the next iteration. thanks!

---
In reply to: [177581409](https://github.com/dotnet/roslyn/pull/25362#discussion_r177581409) [](ancestors = 177581409,176929133)